docs/guide/styles.md

```diff
@@ -1,15 +1,320 @@
+
+### Border 
+
+The [border](../styles/border.md) style draws a border around a widget. To add a border set `syles.border` to a tuple of two values. The first value is the border type, which should be a string. The second value is the border color which will accept any value that works with  [color](../styles/color.md) and [background](../styles/background.md).
+
+The following example adds a border around a widget:
+
+```python title="border01.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/border01.py"
+```
+
+Here is the result:
+
+```{.textual path="docs/examples/guide/styles/border01.py"}
+```
+
+There are many other border types. Run the following from the command prompt to preview them.
+
+
+```bash
+textual borders
+```
+
+### Outline
+
+[Outline](../styles/outline.md) is similar to border and is set in the same way. The difference is that outline will not change the size of the widget, and may overlap the content area. The following examples sets an outline on a widget:
+
+```python title="outline01.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/outline01.py"
+```
+
+Notice how the outline overlaps the text in the widget.
+
+```{.textual path="docs/examples/guide/styles/outline01.py"}
+```
+
+Outline can be useful to emphasize a widget, but be mindful that it may obscure your content.
+
+### Box sizing
+
+When you set padding or border it reduces the size of the widget's content area. In other words, setting padding or border won't change the width or height of the widget.
+
+This is generally desirable when you arrange things on screen as you can add border or padding without breaking your layout. Occasionally though you may want to keep the size of the content area constant and grow the size of the widget to fit padding and border. The [box-sizing](../styles/box_sizing.md) style allows you to switch between these two modes.
+
+If you set `box_sizing` to `"content-box"` then space required for padding and border will be added to the widget dimensions. The default value of `box_sizing` is `"border-box"`. Compare the box model diagram for content box to the previous box model diagram.
+
+<div class="excalidraw">
+--8<-- "docs/images/styles/content_box.excalidraw.svg"
+</div>
+
+The following example creates two widgets which have a width of 30, a height of 6, and a border and padding of 1. The second widget has `box_sizing` set to `"content-box"`.
+
+```python title="box_sizing01.py" hl_lines="33"
+--8<-- "docs/examples/guide/styles/box_sizing01.py"
+```
+
+The padding and border of the first widget is subtracted from the height leaving only 2 lines in the content area. The second widget also has a height of 6, but the padding and border adds additional height so that the content area remains 6 lines. 
+
+```{.textual path="docs/examples/guide/styles/box_sizing01.py"}
```

Comment:
Aside/suggestion: The code snippet here:
```python
        self.widget1.styles.background = "purple"
        self.widget2.styles.background = "darkgreen"
        self.widget1.styles.width = 30
        self.widget2.styles.width = 30
        self.widget2.styles.height = 6
        self.widget1.styles.height = 6
        self.widget2.styles.height = 6
        self.widget1.styles.border = ("heavy", "white")
        self.widget2.styles.border = ("heavy", "white")
        self.widget1.styles.padding = 1
        self.widget2.styles.padding = 1
        self.widget2.styles.box_sizing = "content-box"
```

Makes me wish for an API like:

```python
self.widget1.styles.update(
    background="purple",
    width=30,
    height=6,
    padding=1,
)
```

Inline styles are definitely more verbose than they need to be atm, which is good for selling CSS, but I think both stories can be great 😄